packages/components/organisms/f-checkout/src/components/Checkout.vue

```diff
@@ -281,32 +285,13 @@ export default {
          */
         async submitCheckout () {
             try {
-                const checkoutData = {
-                    customer: {
-                        firstName: this.customer.firstName,
-                        lastName: this.customer.lastName,
-                        phoneNumber: this.customer.mobileNumber,
-                        dateOfBirth: ''
-                    },
-                    fulfilment: {
-                        time: this.time,
-                        location: {
-                            ...(this.isCheckoutMethodDelivery ? {
-                                address: this.address
-                            } : {}),
-                            geolocation: null
-                        }
-                    },
-                    notes: this.notes
-                };
-
                 if (!this.isLoggedIn) {
                     await this.setupGuestUser();
                 }
 
                 await this.patchCheckout({
                     url: `checkout/${this.tenant}/${this.checkoutId}`,
-                    data: checkoutData,
+                    data: mapCheckoutRequest(this),
```

Comment:
What do we think of this? Would people rather I create a new object like this, rather than passing `this` in in its entirety?
```javascript
data: mapCheckoutRequest({
    address: this.address,
    customer: this.customer,
    isCheckoutMethodDelivery: this.isCheckoutMethodDelivery,
    time: this.time,
    userNote: this.userNote
})
```

Reply:
I decided this was probably the better approach even if it's a few more lines.